### ISSUE TITLE
NotImplemented exception thrown on AES unzip.

### ISSUE THREAD

**Issue by BertusVanZyl** [icsharpcode/SharpZipLib#188]
### Steps to reproduce
```.cs
using (FileStream fs = File.OpenRead(fileName))
{
    var zf = new ZipFile(fs);
    zf.Password = "aaa";
    foreach (ZipEntry zipEntry in zf)
    {
        if (!zipEntry.IsFile)
        {
            continue; // Ignore directories
        }
        
        byte[] buffer = new byte[zipEntry.Size];
        try
        {
            using (Stream zipStream = zf.GetInputStream(zipEntry))
            {
                zipStream.Read(buffer, 0, buffer.Length);
            }
        }
        catch (Exception ex)
        {
            //NotImplemented thrown on dispose from using block.
        }
    }
}
```

### Expected behavior
The bytes are read from the stream successfully, so the functionality IS implemented. The exception should not be thrown.

### Actual behavior
NotImplemented exception is thrown.

### Version of SharpZipLib
0.86.0 nuget package.

### Obtained from (place an x between the brackets for all that apply)
- Package installed using:
 - NuGet

**Comment by FrikkinLazer** [comment]
Here is an example zip. the password is "aaa'.

[aaa.zip](https://github.com/icsharpcode/SharpZipLib/files/1265994/aaa.zip)

**Comment by wsloth** [comment]
I have this same problem with the following code:

            var zipFile = new ZipFile(zipStream)
            {
                Password = "Some Encryption Password"
            };

            foreach (ZipEntry entry in zipFile)
            {
                if (!entry.IsFile) continue;

                string content;
                var zipInputStream = zipFile.GetInputStream(entry);

                // Read file contents
                using (var s = new StreamReader(zipInputStream, Encoding.UTF8))
                {
                    content = s.ReadToEnd(); // The file content is properly read without problems
                } // exception is thrown here
            }

It seems to me the exception occurs here: https://github.com/icsharpcode/SharpZipLib/blob/be53259f188e39c0965dfe8d36448ddbd706403b/src/ICSharpCode.SharpZipLib/Encryption/ZipAESTransform.cs#L157

**Comment by stealthrabbi** [comment]
So what should someone do, given this bug? I can get the input stream and read from it. But we can't close it. Should we call it and catch the exception, or just leave it open? If we're unzipping a large file, I'm concerned about leaving all those stream open. Or, is it sufficient to close the contraining stream (e.g. MemoryStream) that is passed to ZipFile ?

**Comment by piksel** [comment]
I don't know. This is planned to be fixed for v1.1, and is currently listed as not working as per #35.

I have not been able to work on it for a while though.

**Comment by BertusVanZyl** [comment]
This has been fixed in 1.2.0. I don't know in which version it was fixed.

**Comment by piksel** [comment]
It was fixed in [v1.1](https://github.com/icsharpcode/SharpZipLib/wiki/Release-1.1), thanks for updating (closing) the issue!
